airflow-core/tests/unit/utils/test_file.py

```diff
@@ -189,6 +189,68 @@ def test_might_contain_dag_with_default_callable(self):
 
         assert file_utils.might_contain_dag(file_path=file_path_with_dag, safe_mode=True)
 
+    def test_airflowignore_negation_unignore_subfolder_file_glob(tmp_path):
+        """Ensure negation rules can unignore a subfolder and a file inside it when using glob syntax.
+
+        Patterns:
+          *                     -> ignore everything
+          !subfolder/           -> unignore the subfolder (must match directory rule)
+          !subfolder/keep.py    -> unignore a specific file inside the subfolder
```

Comment:
I'm surprised that `subfolder/drop.py` is ignored. Could we add a reference to the doc that indicates .gitignores or similar files work this way?